sklearn/metrics/pairwise.py

```diff
@@ -646,19 +647,33 @@ def pairwise_distances_argmin_min(
     """
     X, Y = check_pairwise_arrays(X, Y)
 
-    if metric_kwargs is None:
-        metric_kwargs = {}
-
     if axis == 0:
         X, Y = Y, X
 
-    indices, values = zip(
-        *pairwise_distances_chunked(
-            X, Y, reduce_func=_argmin_min_reduce, metric=metric, **metric_kwargs
+    if metric_kwargs is None:
+        metric_kwargs = {}
+
+    if ArgKmin.is_usable_for(X, Y, metric):
+        values, indices = ArgKmin.get_for(
+            X=X, Y=Y, k=1, metric=metric, metric_kwargs=metric_kwargs
+        ).compute(strategy="auto", return_distance=True)
+        values = np.ndarray.flatten(values)
+        indices = np.ndarray.flatten(indices)
+    else:
+        # TODO: support sparse matrices
+        # When ArgKmin is not supported and when the
+        # user asked for "fast_sqeuclidean", we need to
+        # revert to "euclidean"
+        if metric == "fast_sqeuclidean":
+            metric = "euclidean"
```

Comment:
This is not always equivalent depending on `metric_kwargs["squared"]`: we need to remember to square the result `min` distance values in `pairwise_distances_argmin_min`.

Reply:
The pro of `fast_sqeuclidean` "metric" is that it is way faster but users might be interested in getting euclidean distances if needed.

Euclidean distances are currently returned when:
```python
    metric="fast_sqeuclidean',
    return_distance=True,
```
but it is inconsistent with the naming (`fast_sqeuclidean` solely uses squared euclidean distances for computational reasons).

Would it makes sense I guess we could have two strategies `fast_squared_euclidean` and `fast_euclidean` both using `gemm` but returning the appropriate distances?

Reply:
Yes I believe so.